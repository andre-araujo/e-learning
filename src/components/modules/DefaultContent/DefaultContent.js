import React from 'react';
import { string } from 'prop-types';

import {
  Container,
  Title,
  Description,
} from './DefaultContent.styles';
import { InternalLink } from '../../elements';

function DefaultContent({
  title,
  description,
  href,
  linkText,
}) {
  return (
    <Container>
      <Title>
        {title}
      </Title>
      <Description>
        {description}
      </Description>
      <InternalLink href={href}>
        {linkText}
      </InternalLink>
    </Container>
  );
}

DefaultContent.defaultProps = {
  title: '',
};

DefaultContent.propTypes = {
  title: string,
};

export default DefaultContent;
