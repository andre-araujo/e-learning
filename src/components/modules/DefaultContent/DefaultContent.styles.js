import styled from 'styled-components';

import Wrapper from '../../elements/Wrapper';

export const Container = styled(Wrapper)`
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 400px;
  height: calc(100vh - 72px);
  text-align: center;
  flex-direction: column;
`;

export const Title = styled.h1`
  font-size: 2rem;
  margin-bottom: 10px;
`;

export const Description = styled.p`
  font-size: 1rem;
  margin-bottom: 30px;
`;
