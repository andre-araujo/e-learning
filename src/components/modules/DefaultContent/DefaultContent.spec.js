import React from 'react';
import { shallow } from 'enzyme';

import DefaultContent from './DefaultContent';

describe('DefaultContent component', () => {
  it('should match snapshot', () => {
    expect(shallow(<DefaultContent />)).toMatchSnapshot();
  });
});
