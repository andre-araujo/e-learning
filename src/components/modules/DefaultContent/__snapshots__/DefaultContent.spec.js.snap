// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultContent component should match snapshot 1`] = `
<Styled(Wrapper)>
  <styled.h1 />
  <styled.p />
  <InternalLink
    gap={0}
    href={null}
  />
</Styled(Wrapper)>
`;
